core/internal/features_test.go

```diff
@@ -65,67 +80,55 @@ func TestIntegration_HttpRequestWithHeaders(t *testing.T) {
-	unconfirmedReceipt := (*types.Receipt)(nil)
-	confirmedReceipt := &types.Receipt{
-		TxHash:      attempt1Hash,
-		BlockNumber: big.NewInt(confirmed),
-	}
+	rpcClient.On("EthSubscribe", mock.Anything, mock.Anything, "newHeads").
+		Run(func(args mock.Arguments) { chchNewHeads <- args.Get(1).(chan<- *models.Head) }).
+		Return(sub, nil)
+	rpcClient.On("CallContext", mock.Anything, mock.Anything, "eth_getBlockByNumber", mock.Anything, false).
+		Run(func(args mock.Arguments) { *args.Get(1).(**models.Head) = cltest.Head(inLongestChain) }).
+		Return(nil)
 
-	eth.Context("app.Start()", func(eth *cltest.EthMock) {
-		eth.RegisterSubscription("newHeads", newHeads)
-		eth.Register("eth_getTransactionCount", `0x100`) // TxManager.ActivateAccount()
-	})
-	assert.NoError(t, app.StartAndConnect())
+	gethClient.On("ChainID", mock.Anything).Return(config.ChainID(), nil)
+	gethClient.On("BalanceAt", mock.Anything, mock.Anything, mock.Anything).Maybe().Return(oneETH.ToInt(), nil)
+	gethClient.On("BlockByNumber", mock.Anything, big.NewInt(inLongestChain)).Return(cltest.BlockWithTransactions(), nil)
 
-	eth.Context("ethTx.Perform()#1 at block 23456", func(eth *cltest.EthMock) {
-		eth.Register("eth_sendRawTransaction", attempt1Hash) // Initial tx attempt sent
-		eth.Register("eth_getTransactionReceipt", unconfirmedReceipt)
-	})
-	j := cltest.CreateHelloWorldJobViaWeb(t, app, mockServer.URL)
-	jr := cltest.WaitForJobRunToPendOutgoingConfirmations(t, app.Store, cltest.CreateJobRunViaWeb(t, app, j))
+	gethClient.On("SendTransaction", mock.Anything, mock.Anything).
+		Run(func(args mock.Arguments) {
+			tx, ok := args.Get(1).(*types.Transaction)
+			require.True(t, ok)
+			gethClient.On("TransactionReceipt", mock.Anything, mock.Anything).
+				Return(&types.Receipt{TxHash: tx.Hash(), BlockNumber: big.NewInt(confirmed)}, nil)
+		}).
+		Return(nil).Once()
 
-	cltest.WaitForTxAttemptCount(t, app.Store, 1)
+	sub.On("Err").Return(nil)
+	sub.On("Unsubscribe").Return(nil).Maybe()
 
-	jr.ObservedHeight = (*utils.Big)(confirmedReceipt.BlockNumber)
-	require.NoError(t, app.Store.SaveJobRun(&jr))
+	assert.NoError(t, app.StartAndConnect())
 
-	eth.Context("ethTx.Perform()#4 at block 23465", func(eth *cltest.EthMock) {
-		eth.Register("eth_getTransactionReceipt", confirmedReceipt) // confirmed for gas bumped txat
-	})
-	newHeads <- cltest.Head(safe) // 23465
+	newHeads := <-chchNewHeads
 
-	cltest.WaitForTxAttemptCount(t, app.Store, 1)
+	j := cltest.CreateHelloWorldJobViaWeb(t, app, mockServer.URL)
+	jr := cltest.WaitForJobRunToPendOutgoingConfirmations(t, app.Store, cltest.CreateJobRunViaWeb(t, app, j))
+	cltest.WaitForEthTxAttemptCount(t, app.Store, 1)
 
-	cltest.WaitForJobRunToComplete(t, app.Store, jr)
+	// Do the thing
+	newHeads <- cltest.Head(safe)
 
-	eth.EventuallyAllCalled(t)
+	cltest.WaitForJobRunToComplete(t, app.Store, jr)
 }
 
-func TestIntegration_FeeBump(t *testing.T) {
+func TestIntegration_FeeBump_LegacyTXM(t *testing.T) {
```

Comment:
Some of these tests seem like they should have equivalent BPTXM versions, but they don't stand out to me(searching for "FeeBump"). Is there a BPTXM equivalent somewhere?

Reply:
the BPTXM fee bump logic is well tested in `eth_broadcaster_test.go`, while the the `TestIntegration_NonceManagement_firstRunWithExistingTxs` test doesn't really make sense in the BPTXM world (because there is no longer an initial check for tx count). These will be removed as part of the legacy TXM removal.